generic.go

```diff
@@ -3,8 +3,27 @@ package testcontainers
 import (
 	"context"
 	"fmt"
+	"sync"
 )
 
+const (
```

Comment:
Just a personal opinion, not a blocker: wdyt about moving all this new code to a `parallel.go`, or `generic_parallel.go` file, so that all the concerns about parallelisation are all together in one same file? I'd like to listen to your thoughts, and if you see it convenient to keep the code next to the generic implementation, what would it be the motivation?